Sources/FoundationMacros/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift open source project
+##
+## Copyright (c) 2024 Apple Inc. and the Swift project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+# SwiftSyntax Dependency
+include(FetchContent)
+find_package(SwiftSyntax)
+if(NOT SwiftSyntax_FOUND)
+    # If building at desk, check out and link against the SwiftSyntax repo's targets
+    FetchContent_Declare(SwiftSyntax
+        GIT_REPOSITORY https://github.com/apple/swift-syntax.git
+        GIT_TAG 510.0.0)
```

Comment:
Can't do much about it at the moment, but it's unfortunate that this tag doesn't have the changes for picking up the `SWIFT_HOST_MODULE_TRIPLE` automatically. So just as a heads up, you're likely to get folks forgetting to set that and asking about an the error message:

```
CMake Error at build/_deps/swiftsyntax-src/CMakeLists.txt:80 (message):
  Host module triple required
```

The correct response will be to run `swiftc -print-target-info` manually and pull out the triple under `target` > `moduleTriple` and pass that to their CMake invocation e.g. `cmake -G 'Ninja' -DSWIFT_HOST_MODULE_TRIPLE=x86_64-unknown-linux-gnu ...`.